src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -320,7 +320,9 @@ export class ExtensionsListView extends CollapsibleView {
 				}
 
 				return this.extensionsWorkbenchService.queryGallery(assign(options, { names, pageSize: names.length }))
-					.then(pager => new PagedModel(pager));
+					.then(pager => this.addRecommendationInfo(new PagedModel(pager || []), {
+						source: 'fileType'
+					}));
```

Comment:
@ramya-rao-a I am not convinced with `updateTelemetryData` method on `Extension` because there can be multiple instances of `Extension` for same gallery extension (SearchedExtension, InstallingExtension, InstalledExtension). This will fix the current problem. But in future, if we want to report the similar telemetry for successful installs, this might not work.

If I see both the issues (this and mine), what we are doing is to report query related telemetry for eg., source of the query, query text, result index. The problem we have is we parse Search text at two places and Parsed query in extension gallery service does not know about the real query. 

Ideally, we could move all parsing to one place (may be to GalleryService). But for now we can send `userSearchText` as a query option from where we can compute telemetry data.

What do you think?